File: src/lever.ts
import { Coord, Dir } from "./coord";

export class Lever {
    private pos: Coord;
    private dir: Dir;
    constructor(pos: Coord, dir: Dir) {
        this.pos = pos;
        this.dir = dir;
    }

    is_on() {
        return `block ${this.pos.raw()} minecraft:lever[powered=true]`;
    }

    is_off() {
        return `block ${this.pos.raw()} minecraft:lever[powered=false]`;
    }

    set_on() {
        return this.set(true);
    }

    set_off() {
        return this.set(false);
    }

    set(on: boolean) {
        return `setblock ${this.pos.raw()} minecraft:lever[facing=${this.dir},powered=${on}]`;
    }
}

File: mcdsl/src/main/kotlin/com/zacklukem/mcdsl/blocks/Lever.kt
package com.zacklukem.mcdsl.blocks

import com.zacklukem.mcdsl.util.Coord
import com.zacklukem.mcdsl.util.Dir
import com.zacklukem.mcdsl.util.Condition
import com.zacklukem.mcdsl.util.con

class Lever(private val pos: Coord, private val dir: Dir) {
    fun isOn(): Condition {
        return con("block $pos minecraft:lever[powered=true]")
    }

    fun isOff(): Condition {
        return con("block $pos minecraft:lever[powered=false]")
    }

    fun setOn(): String {
        return set(true)
    }

    fun setOff(): String {
        return set(false)
    }

    private fun set(on: Boolean): String {
        return "setblock $pos minecraft:lever[facing=$dir,powered=$on]"
    }
}
